pkg/cloud/vsphere/context/cluster_context.go

```diff
@@ -309,13 +309,18 @@ func (c *ClusterContext) ControlPlaneEndpoint() (string, error) {
 		return "", errors.Wrap(err, "error creating machine context while searching for control plane endpoint")
 	}
 
-	if ipAddr := machineCtx.IPAddr(); ipAddr != "" {
-		controlPlaneEndpoint := net.JoinHostPort(ipAddr, strconv.Itoa(int(machineCtx.BindPort())))
-		machineCtx.Logger.V(2).Info("got control plane endpoint from machine", "control-plane-endpoint", controlPlaneEndpoint)
-		return controlPlaneEndpoint, nil
+	ipAddr, err := machineCtx.IPAddr()
+	if err != nil {
+		return "", errors.Wrap(err, "error getting first IP address for machine")
+	}
+
+	if ipAddr == "" {
```

Comment:
Hi @andrewsykim,

This shouldn't be required since `ipAddr` should never empty without an error.